Tools/peg_generator/pegen/parser.py

```diff
@@ -21,7 +21,7 @@ def logger(method: F) -> F:
     """
     method_name = method.__name__
 
-    def logger_wrapper(self: P, *args: object) -> T:
+    def logger_wrapper(self: P, *args: object) -> Any:
```

Comment:
Since we're touching this line anyway, you could change `P` here to `Parser`: using a `TypeVar` here is meaningless:

```suggestion
    def logger_wrapper(self: Parser, *args: object) -> Any:
```